### PR TITLE
UI Simplification: Add Design panel button to replace background media

### LINKED ISSUE #10108: UI Simplification: Allow changing background media from the panel
## Feature Description
See [Figma](https://www.figma.com/file/dc9xrVlRhQq4KTE9WQzU1R/_Editor---Sprint-Feature-Level-Designs_?node-id=11122%3A447721)

We will be using the design with button "Replace".

Currently, the background media can be replaced by dragging a new image onto the Page, or by clicking "Set as background" while having another media element selected.

We should allow doing it directly from the Page Background panel as well.

- [ ] Add a new "Replace" button.
- [ ] This button will open Media library to replace the background media

### PR DESCRIPTION
**Pull request by miina**
## Summary
Refactors code to have a separate Upload button component.
Uses this to allow replacing background media from the Page background panel directly.

- [x] Track button usage
- [x] Tests

## User-facing changes
| Before | After |
| ------ | ----- |
| Screenshot 2022-01-10 at 19 51 47 | Screenshot 2022-01-10 at 19 48 53 |

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Add background media
2. Click "Replace" button in the Page background panel
3. Verify it's possible to both choose media from the library / upload new media
4. Verify the "Upload" button in the Media tab still works as expected.

### Does this PR have a security-related impact?
No

### Does this PR change what data or activity we track or use?
Adds new event `replace_background_media`.

### Does this PR have a legal-related impact?
No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10108

### CODE REVIEW

File: packages/story-editor/src/components/panels/design/pageBackground/test/pageBackground.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      type: 'image',
+      rotationAngle: 0,
+      resource: {
+        id: 'resource/123',
+        src: 'media.jpg',
+        alt: 'Media',
+      },
+    };
+    const { combineElements } = arrange([backgroundImage]);
+    const replaceButton = screen.getByRole('button', { name: 'Replace' });
```
Comment (barklund) [inline packages/story-editor/src/components/panels/design/pageBackground/test/pageBackground.js:127]: This ought to be moved to `fixture.editor.inspector.designPanel.pageBackground.replace`.

File: packages/story-editor/src/components/form/mediaUploadButton.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    ]
+  );
+
+  return (
+    <MediaUpload
+      onSelect={onSelect}
+      onSelectErrorMessage={onSelectErrorMessage}
+      onClose={resetWithFetch}
+      type={allowedMimeTypes}
+      render={renderButton}
```
Comment (barklund) [inline packages/story-editor/src/components/form/mediaUploadButton.js:144]: Not completely sure why this is rendered as a callback and not just a (child) component, but I know that's what it also did before, so I guess that's fine.

**Review by barklund — APPROVED** [review]
Seems to work fine - and nice refactor with the new MediaUploadButton abstraction.

File: packages/story-editor/src/components/panels/design/pageBackground/test/pageBackground.js (hunk 1 above)
Comment (miina) [reply packages/story-editor/src/components/panels/design/pageBackground/test/pageBackground.js:127]: This is a unit test though, not Karma.

**Pull request merged**